contracts/interchain-token-service/src/contract/query.rs

```diff
@@ -34,6 +34,12 @@ pub fn all_its_contracts(deps: Deps) -> Result<Binary, Error> {
     to_json_binary(&contract_addresses).change_context(Error::JsonSerialization)
 }
 
+pub fn its_chains(deps: Deps, filter: Option<msg::ChainConfigFilter>) -> Result<Binary, Error> {
+    let chain_configs =
+        state::load_filtered_chain_configs(deps.storage, filter).change_context(Error::State)?;
```

Comment:
I would just filter here and not add the filter logic into the state module. The implementation in the state module loads all of the chains anyways, and then filters